File: Cpp/leetcode/kadanesAlgo53.js
var maxSubArray = function(nums) {

    let currSum =0 ;
    let maxSum =Number.MIN_SAFE_INTEGER ;
    nums.forEach(i => {
        currSum += i;
        maxSum = Math.max(currSum , maxSum)
        if(currSum < 0) currSum = 0
        
        return maxSum
    });
}

maxSubArray ([-2,1,-3,4,-1,2,1,-5,4]);

File: Cpp/leetcode/kadanesAlgo53.cpp
#include <iostream>
#include <climits>
#include <vector>
using namespace std;
int main(){
     
    vector<int> arr = {-2,1,-3,4,-1,2,1,-5,4};
    
    int currSum=0 ,maxSum = INT_MIN;
    for ( int i : arr ){
        currSum += i;
        maxSum = max(currSum , maxSum );
        if(currSum < 0){
            currSum = 0 ;
        }

    }
    cout << maxSum <<endl;
    return maxSum;
    
}